dali/server/daldap.cpp

```diff
@@ -115,29 +117,36 @@ class CDaliLdapConnection: implements IDaliLdapConnection, public CInterface
             return SecAccess_Full;
 
 
+        Owned<ISecUser> user;
         StringBuffer username;
         StringBuffer password;
         if (udesc)
         {
             udesc->getUserName(username);
             udesc->getPassword(password);
+            user.setown(ldapsecurity->createUser(username));
+            user->setAuthenticateStatus(AS_AUTHENTICATED);  // treat caller passing user as trusted
         }
         else
         {
             DBGLOG("NULL UserDescriptor in daldap.cpp getPermissions('%s')", key);
-        }
 
-        if (0 == username.length())
-        {
+            // If no user was provided, try to use the default user
+            if (disableFilesDefaultUser || filesdefaultuser.isEmpty())
+            {
+                OWARNLOG("Default user missing or disabled, access denied for request %s %s", key, nullText(obj));
+                return SecAccess_None; // no access if no default user or disabled
+            }
+
             username.append(filesdefaultuser);
             decrypt(password, filesdefaultpassword);
-            OWARNLOG("Missing credentials, injecting deprecated filesdefaultuser for request %s %s", key, nullText(obj));
+            OWARNLOG("Missing credentials, injecting deprecated filesdefaultuser (%s) for request %s %s", filesdefaultuser.str(), key,
+                     nullText(obj));
             logNullUser(nullptr);
```

Comment:
minor: should this be above the if (disabledFileDefaultUser) so still useful when option on?

Also, I think it is already called before each call to this function in 1 context (in getPermissionsLDAP), that call should prob. be removed and just have 1 here. Not that it will make much difference as it only logs once per minute anyway.

Reply:
I moved the logNullUser above the disable check. It would continue to be a good idea to have the information while we are analyzing each case. I did see it's called before this method is called as well. I am going assess what logging is best and make appropriate changes. Right now, more is probably better, and at only once per minute we should be OK until those changes are made.